### ISSUE TITLE
False positive on echo

### ISSUE THREAD

**Issue by GoldsteinE** [koalaman/shellcheck#1330]
#### For bugs
- Rule Id: SC2039
- version 0.3.7
- [x] I read the issue's wiki page, e.g. https://github.com/koalaman/shellcheck/wiki/SC2086
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/sh
echo "-e hello"
```

#### Here's what shellcheck currently says:
```In - line 2:
echo "-e hello"
     ^-- SC2039: In POSIX sh, echo flags are not supported.
```

#### Here's what I wanted or expected to see:
No error. "-e hello" is not a flag, it's a string with space.

**Comment by bittorf** [comment]
you can avoid that (and this is common) by using:

echo -- "-e foo"

**Comment by GoldsteinE** [comment]
No, I totally can't.
```
sh-4.3$ echo -- "-e foo"
-- -e foo```
How correctly says shellcheck,
> In POSIX sh, echo flags are not supported
So `-- -e foo` will be outputted literally.

**Comment by bittorf** [comment]
sorry about my mistake, you are correct and this needs a fix.

**Comment by koalaman** [comment]
This is an interesting case.

In POSIX, the command works perfectly fine because no options (not even `--`) shall be recognized. This means that it shouldn't really be silenced by `--`.  The advice should probably be to use `printf` whenever arguments start with a `-`.

In Bash, it works because `"-e hello"` is still arguably a set of flags, but since it contains currently unsupported options (`-hlo `) it's being printed verbatim:

    $ echo "-Eeenenenenenenne" # BATMAN
    (no output)

    $ echo "-Eeenenenenenennex"
    -Eeenenenenenennex

Though of course there is no warning there.

**Comment by koalaman** [comment]
Thanks to @Gandalf- , this now only warns when the argument consists entirely of known `echo` flags (specifically `-eEsn`)
